Add DeepSeek API key support and participant

wojtess opened the pull request:
Added DeepSeek API key support
Introduced DeepSeekParticipant
Updated ModelConfigs with deepseek-chat
Incremented userCount for DeepSeek API
Ensured participants check for availability

wojtess committed:
Add DeepSeek API key support and participant

Added DeepSeek API key support
Introduced DeepSeekParticipant
Updated ModelConfigs with deepseek-chat
Incremented userCount for DeepSeek API
Ensured participants check for availability

rimplesimpledimple commented on conversation/deepseek.go:
```diff
@@ -0,0 +1,201 @@
+	return chatCompletion.Choices[0].Message.Content, nil
+}
+
+func (p *DeepSeekParticipant) manageConversation(messages []openai.ChatCompletionMessage) []openai.ChatCompletionMessage {
```
Looks like you missed checking that we are not exceeding the context window. `manageConversation` is not getting called anywhere.
` manageConversation(messages []openai.ChatCompletionMessage) []openai.ChatCompletionMessage  ` needs to be updated to deepseek struct.

rimplesimpledimple replied:
Please do update readme to include deepseek as well, thanks.

wojtess replied:
Is this good implementation?
```golang
history = append(history, Message{
From:    p.GetName(),
Content: chatCompletion.Choices[0].Message.Content,
})

history = p.manageConversation(history)

return history[len(history)-1].Content, nil
```

rimplesimpledimple replied:
Similar to other implementations:

```
var messagesPayload []messagePayload
	for i := range history {
		var role string
		if strings.Compare(strings.ToLower(history[i].From), "system") == 0 {
			role = "system"
		} else if strings.Compare(strings.ToLower(history[i].From), strings.ToLower(p.GetName())) == 0 {
			role = "assistant"
		} else {
			role = "user"
		}
		messagesPayload = append(messagesPayload, messagePayload{Role: role, Content: history[i].Content})
	}
	
// handle context window
messagesPayload = p.manageConversation(messagesPayload)
		

```

Update manageConversation accordingly.

wojtess committed:
Fixed as proposed in comment

https://github.com/rimplesimpledimple/llm-discuss/pull/2#pullrequestreview-2400042198

rimplesimpledimple reviewed (approved):
Thanks for contributing

wojtess committed:
added config

wojtess committed:
fixed system prompt generation in deepseek

wojtess committed:
added REST support

wojtess commented:
I added other options such as config and REST
@rimplesimpledimple

wojtess commented:
What is status?
